lib/api/apiUtils/bucket/validateReplicationConfig.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Validates that the replication configuration will contain the default
+ * read location as a site if the read and write locations are different
+ * @param {object} config - replication configuration
+ * @param {object} bucket - bucket metadata
+ * @return {boolean} validity of replication configuration with rest endpoint
+ * configuration
+ */
+function validateConfiguration(config, bucket) {
+    const writeLocation = bucket.getLocationConstraint();
+    const readLocation = bucket.getReadLocationConstraint();
+    if (!config || !config.rules) {
+        return false;
+    }
+    const isValid = config.rules.some(rule => {
```

Comment:
Don't we want to check that every rule includes the storage class?

Reply:
My thought is that a valid configuration will contain the preferred read location as a storage class in one of the rules.

Reply:
I can see that there might be an error if the replication rule has a prefix.

Reply:
Great! That's the kind of case I had in mind.